test/unit/org/apache/cassandra/utils/Generators.java

```diff
@@ -314,11 +324,36 @@ public static Gen<ByteBuffer> bytes(int min, int max)
             // to add more randomness, also shift offset in the array so the same size doesn't yield the same bytes
             int offset = (int) rnd.next(Constraint.between(0, MAX_BLOB_LENGTH - size));
 
-            return ByteBuffer.wrap(LazySharedBlob.SHARED_BYTES, offset, size);
+            return handleCases(cases, rnd, offset, size);
         };
+    };
+
+    private enum BBCases { HEAP, READ_ONLY_HEAP, DIRECT, READ_ONLY_DIRECT }
+
+    private static ByteBuffer handleCases(Gen<BBCases> cases, RandomnessSource rnd, int offset, int size) {
+        switch (cases.generate(rnd))
+        {
+            case HEAP: return ByteBuffer.wrap(LazySharedBlob.SHARED_BYTES, offset, size);
+            case READ_ONLY_HEAP: return ByteBuffer.wrap(LazySharedBlob.SHARED_BYTES, offset, size).asReadOnlyBuffer();
+            case DIRECT:
+            {
+                ByteBuffer bb = ByteBuffer.allocateDirect(size);
+                bb.put(LazySharedBlob.SHARED_BYTES, offset, size);
+                bb.flip();
+                return bb;
```

Comment:
when I said you should avoid copy/paste and make this a method, I mean 

```
ByteBuffer bb = ByteBuffer.allocateDirect(size);
                 bb.put(LazySharedBlob.SHARED_BYTES, offset, size);
                 bb.flip();
```

and not this new `handleCases`...

Reply:
the new handleCases helps with readability so my plan is to keep that method and add "ByteBuffer bb = directBufferFromSharedBlob(offset, size);" to address the copy/paste code. Any opinions on removing or keeping the method?